src/Course/StateT.hs

```diff
@@ -241,31 +217,66 @@ instance Functor f => Functor (OptionalT f) where
 -- >>> runOptionalT $ OptionalT (Full (+1) :. Full (+2) :. Nil) <*> OptionalT (Full 1 :. Empty :. Nil)
 -- [Full 2,Empty,Full 3,Empty]
 instance Monad f => Applicative (OptionalT f) where
-  pure =
-    error "todo: Course.StateT pure#instance (OptionalT f)"
-  (<*>) =
-    error "todo: Course.StateT (<*>)#instance (OptionalT f)"
+  pure :: a -> OptionalT f a
+  pure x = OptionalT (pure (Full x))
+
+  (<*>) :: OptionalT f (a -> b) -> OptionalT f a -> OptionalT f b
+  (<*>) optTF optTX =
+    OptionalT (
+      runOptionalT optTF
+      >>= onFull (\f ->         -- :: (a -> b)
+        runOptionalT optTX
```

Comment:
nice... I used the unwrapping "run" functions hardly at all and I think that was a mistake...